### PR TITLE
Allow unnamed assembly in IVT check

### LINKED ISSUE #29253: ArgumentNullException binding inaccessible type
`csc /t:library a.cs`:
```c#
class A { }
```
`csc /r:a.dll b.cs`:
```c#
class B
{
    static void Main()
    {
        new A();
    }
}
```
```
Unhandled Exception: System.AggregateException: One or more errors occurred. ---> System.ArgumentNullException: Value cannot be null.
Parameter name: key
   at System.ThrowHelper.ThrowArgumentNullException(ExceptionArgument argument)
   at System.Collections.Generic.Dictionary`2.FindEntry(TKey key)
   at System.Collections.Generic.Dictionary`2.TryGetValue(TKey key, TValue& value)
   at Microsoft.CodeAnalysis.PEAssembly.GetInternalsVisibleToPublicKeys(String simpleName)
   at Microsoft.CodeAnalysis.CSharp.Binder.<CheckViability>g__IsBadIvtSpecification|564_0(<>c__DisplayClass564_0& )
   at Microsoft.CodeAnalysis.CSharp.Binder.CheckViability(Symbol symbol, Int32 arity, LookupOptions options, TypeSymbol accessThroughType, Boolean diagnose, HashSet`1& useSiteDiagnostics, ConsList`1 basesBeingResolved)
```

### PR DESCRIPTION
**Pull request by cston**
Fixes #29253.

### CODE REVIEW

**Comment by cston** [comment]
@dotnet/roslyn-compiler please review.

**Comment by cston** [comment]
@dotnet-bot test windows_release_vs-integration_prtest

File: src/Compilers/CSharp/Test/Semantic/Semantics/AccessCheckTests.cs [hunk 1]
```diff
@@ -1243,5 +1243,74 @@ public Class1()
 }";
             CompileAndVerify(text).VerifyDiagnostics();
         }
+
+        [WorkItem(29253, "https://github.com/dotnet/roslyn/issues/29253")]
+        [Fact]
+        public void InaccessibleToUnnamedExe_01()
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Test/Semantic/Semantics/AccessCheckTests.cs]: >InaccessibleToUnnamedExe_01 [](start = 20, length = 27)

Please add similar tests for VB

**Comment by cston** [comment]
@dotnet-bot test windows_debug_vs-integration_prtest

**Comment by cston** [comment]
@dotnet-bot test windows_debug_spanish_unit32_prtest

**Comment by cston** [comment]
@dotnet-bot test windows_release_vs-integration_prtest

**Comment by jaredpar** [comment]
Approved to merge

**Comment by cston** [comment]
@dotnet-bot test windows_debug_spanish_unit32_prtest

**Comment by cston** [comment]
Following up with @AlekseyTs, it looks like the one place where we could check `Compilation.AssemblyName` for `null` is when `AssemblyName` is used as a `Diagnostic` [argument](https://github.com/dotnet/roslyn/blob/master/src/Compilers/CSharp/Portable/Emitter/EditAndContinue/EmitHelpers.cs#L52). But it looks like `DiagnosticInfo.GetMessage()` should handle `null` values.

**Pull request merged**